src/43F0.c

```diff
@@ -74,9 +74,128 @@ HeapNode* _heap_create(s32* addr, u32 size) {
     }
 }
 
-INCLUDE_ASM(s32, "43F0", _heap_malloc);
+#define _heap_alloc_and_update_id(node) { \
+    u16 HeapEntryID = D_NEXT_HEAP_MALLOC_ID; \
+    node->allocated = 1; \
+    D_NEXT_HEAP_MALLOC_ID = HeapEntryID + 1; \
+    node->entryID = HeapEntryID; \
+}
+
+void* _heap_malloc(HeapNode* head, u32 size) {
+    HeapNode* nextHeapNode;
+    HeapNode* pPrevHeapNode = 0;
+    u32 newBlockSize;
+    u32 curBlockLength;
+    HeapNode* curHeapNode;
+    u32 smallestBlockFound;
+
+    //must allocate 16 bytes or more at minimum or fail
+    size = ALIGN16(size);
+    if (size == 0) {
+        return (s32 *)pPrevHeapNode;
```

Comment:
Is this cast required? Additionally, does it not match as `return NULL`?

Reply:
Cast is not required, it can return null and changed. Was a left over from getting the source to line up early on